lua/indent_blankline/init.lua

```diff
@@ -124,23 +125,39 @@ local refresh = function()
     local space_char_blankline_highlight_list = v("indent_blankline_space_char_blankline_highlight_list")
     local space_char_blankline = v("indent_blankline_space_char_blankline")
 
-    local space_character = vim.opt.listchars:get().space or " "
-    -- source tab chars in a way that works with UTF-8 characters
-    local tab_characters
-    if vim.opt.listchars:get().tab then
-        tab_characters = vim.fn.split(vim.opt.listchars:get().tab, "\\zs")
+    local list_chars = {}
+    -- No need to check for disable_with_list as this part would never be executed if "true" && nolist
+    if vim.opt.list:get() then
+        -- list is set, get listchars
+        local tab_characters
+        local space_character = vim.opt.listchars:get().space or " "
+        if vim.opt.listchars:get().tab then
+            -- tab characters can be any UTF-8 character, Lua 5.1 cannot handle this without external libraries
+            tab_characters = vim.fn.split(vim.opt.listchars:get().tab, "\\zs")
+        else
+            tab_characters = {}
+        end
+        list_chars = {
+            space_char = space_character,
+            trail_char = vim.opt.listchars:get().trail or space_character,
+            lead_char = vim.opt.listchars:get().lead or space_character,
+            tab_char_start = tab_characters[1] or space_character,
+            tab_char_fill = tab_characters[2] or space_character,
+            tab_char_end = tab_characters[3],
+            eol_char = vim.opt.listchars:get().eol
+        }
     else
-        tab_characters = {}
+        -- nolist is set, replace all listchars with empty space
+        list_chars = {
+            space_char = " ",
+            trail_char = " ",
+            lead_char = " ",
+            tab_char_start = " ",
+            tab_char_fill = " ",
+            tab_char_end = " ",
+            eol_char = char
```

Comment:
actually this might be confusing, even if it does the same thing.
Just set it to `""`

Reply:
Haha, I forgot already that there is a check to see of that char exists.
`""` won't work however, because Lua interprets even a zero-length string as `true`. I will set it explicitly to `nil`so it doesn't look like it was forgotten.

Same with `tab_char_end`, setting it to `nil` will result in the shorter branch inside `find_indent()`. That optimization will speed up the code by at least 10x 😎

Reply:
:rocket: